serve/mlc_serve/engine/base.py

```diff
@@ -1,11 +1,45 @@
 from dataclasses import dataclass, field
 from enum import Enum
-from typing import Optional
-
+from typing import Optional, List
+import json
+import inspect
 from .sampling_params import SamplingParams, SamplingType
 
 RequestId = str
 
+@dataclass
+class MLCServeEngineConfig:
+    # The maximum number of tokens in the batch.
+    # TODO(@sunggg): figure out better defaults
+    # from sync engine
+    # max_batched_tokens: int = 2560,
+    # min_decode_steps: int = 100,
+    # max_decode_steps: int = 300,
+    # prompt_allocate_ratio: float = 2.0,
+
+    # from staging engine
+    # max_batched_tokens: int = 2560,
+    # min_decode_steps: int = 32,
+    # max_decode_steps: int = 48,
+    # prompt_allocate_ratio: float = 2.0,
+    
+    use_staging_engine: bool = True
+    max_batched_tokens: int = 2560
+    max_input_len: int = 2560
```

Comment:
Might want to rename to 1max_input_tokens_len` or something like that just so its clear that its number of tokens, having multiple "max" values can get confusing here.

Reply:
previously, it was `max_num_batched_tokens` but I shortened it. If this sounds good to you, I can revert this change.

Reply:
There is no reason to stick to 2560. I have been using it since this was the default vllm was using until recently. We can bump it to 4k for example.

Reply:
resolved.